### PR TITLE
Load sample analysis into trending db

### PR DESCRIPTION
**Pull request by hassanfa**
This PR:

1. updates `vogue load  analysis` with three new cli options:
```
  -c, --analysis-case TEXT        The case that this sample belongs to. It can
                                  be specified multiple times.  [required]
  -w, --analysis-workflow TEXT    Analysis workflow used.  [required]
  --workflow-version TEXT         Analysis workflow used.  [required]
```
2. Changes the structure of a sample's analysis results to the following (assuming the case name is: case_1 and workflow is mip version 6.5):
```
{case: {case_1: {workflow: {mip : {workflow_version: 6.5, analysis_dict} } } } }
```
*analysis_dict is constructed in `build/analysis.py`

3.`analysis` collection is renamed to `analysis_sample`

To test this PR:
1. `git checkout analysis_case_workflow`
2. set the following environment variables:
```
export FLASK_DEBUG=0
export MONGO_URI=mongodb://localhost:${port_for_mongo_db}
export MONGO_DBNAME=${dbname_within_mongod}
export VOGUE_SECRET_KEY=${your_secret_key}
```
3. Run vogue to load into DB:
```
vogue load analysis \
    --sample-id test_sample_id \
    --analysis-config tests/fixtures/valid_multiqc.json \
    --analysis-case test_case_name \
    --analysis-workflow awesome_workflow_2 \
    --workflow-version 54
```
and an update the sample with extra analysis from another workflow:
```
vogue load analysis \
    --sample-id test_sample_id \
    --analysis-config tests/fixtures/valid_multiqc.json \
    --analysis-case test_case_name \
    --analysis-workflow awesome_workflow \
    --workflow-version 3.14159.2.71828
```

4. Voilà!
image

### CODE REVIEW

**Comment by hassanfa** [comment]
Tests will fail. But being urgent and what not, they'll be added at a more convenient future.  :-)

**Comment by mayabrandi** [comment]
Sweet Hassan. Im looking it through now.

File: vogue/adapter/plugin.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def setup(self, db_name : str):
         self.db = self.client[db_name]
         self.db_name = db_name
         self.sample_collection = self.db.sample
-        self.analysis_collection = self.db.analysis
+        self.analysis_collection_sample = self.db.analysis_sample
```
Comment (mayabrandi) [inline vogue/adapter/plugin.py]: I think we decided the oposit ..
```suggestion
        self.sample_analysis_collection = self.db.sample_analysis
```

File: vogue/adapter/plugin.py [hunk 2]
```diff
@@ -86,22 +86,22 @@ def delete_sample(self):
     def add_or_update_analysis(self, analysis_result: dict):
         """Functionality to add or update analysis sample"""
         lims_id = analysis_result['_id']
-        update_result = self.db.analysis.update_one({'_id' : lims_id}, {'$set': analysis_result}, upsert=True)
+        # pop _id key to make pushing easier
+        analysis_result.pop('_id')
+        update_result = self.db.analysis_sample.find_one({'_id': lims_id})
```
Comment (mayabrandi) [inline vogue/adapter/plugin.py]:
```suggestion
        update_result = self.db.sample_analysis.find_one({'_id': lims_id})
```

File: vogue/adapter/plugin.py [hunk 3]
```diff
@@ -86,22 +86,22 @@ def delete_sample(self):
...
         else:
-            LOG.info("No analysis updates for sample %s.", lims_id)
+            self.db.analysis_sample.update_one({'_id' : lims_id}, 
+                    {'$push': analysis_result, '$set': {'updated': dt.today()}}, upsert=True)
+            LOG.info("Updated analysis for sample %s.", lims_id)
 
     def analysis(self, analysis_id: str):
         """Functionality to get analyses results"""
-        return self.analysis_collection.find_one({'_id':analysis_id})
+        return self.analysis_collection_sample.find_one({'_id':analysis_id})
```
Comment (mayabrandi) [inline vogue/adapter/plugin.py]:
```suggestion
        return self.sample_analysis_collection.find_one({'_id':analysis_id})
```

File: vogue/adapter/plugin.py [hunk 4]
```diff
@@ -86,22 +86,22 @@ def delete_sample(self):
...
+            self.db.analysis_sample.update_one({'_id' : lims_id}, 
+                    {'$set': {**analysis_result, **{'added': dt.today()}}},  upsert=True)
             LOG.info("Added analysis sample %s.", lims_id)
-        elif update_result.modified_count:
-            self.db.analysis.update_one({'_id' : lims_id}, 
-                {'$set': {'updated': dt.today()}})
-            LOG.info("Updated analysis for sample %s.", lims_id)
         else:
-            LOG.info("No analysis updates for sample %s.", lims_id)
+            self.db.analysis_sample.update_one({'_id' : lims_id},
```
Comment (mayabrandi) [inline vogue/adapter/plugin.py]:
```suggestion
            self.db.sample_analysis.update_one({'_id' : lims_id}, 
```

File: vogue/adapter/plugin.py [hunk 5]
```diff
@@ -86,22 +86,22 @@ def delete_sample(self):
...
-        update_result = self.db.analysis.update_one({'_id' : lims_id}, {'$set': analysis_result}, upsert=True)
+        # pop _id key to make pushing easier
+        analysis_result.pop('_id')
+        update_result = self.db.analysis_sample.find_one({'_id': lims_id})
 
-        if not update_result.raw_result['updatedExisting']:
-            self.db.analysis.update_one({'_id' : lims_id}, 
-                {'$set': {'added': dt.today()}})
+        if update_result is None:
+            self.db.analysis_sample.update_one({'_id' : lims_id},
```
Comment (mayabrandi) [inline vogue/adapter/plugin.py]:
```suggestion
            self.db.sample_analysis.update_one({'_id' : lims_id}, 
```

**Comment by hassanfa** [comment]
@mayabrandi please have a look on the new document structure. This is what we discussed on avoiding arrays. I also removed `analysis` key name from case object. Now there is a new key called: `workflows` which will make it easier to query if there an analysis for current sample.
Here is how it looks like now:
image

**Comment by hassanfa** [comment]
Also, if we are ok with the structure, I can go ahead and use actual samples.

**Pull request merged**